### ISSUE TITLE
stat_cor to provide low p-value (and not p < 2.2e-16)

### ISSUE THREAD

**Issue by pforny** [kassambara/ggpubr#441]
Hi,
I really enjoy this package and very frequently use the stat_cor function within ggplot. I have one problem though: I would like to display very low p-values and not p < 2.2e-16. Is there a way to work around this one line of code (line 221 currently) in the stat_cor function?

` label <- ifelse(x < 2.2e-16, "p < 2.2e-16", paste0("p = ", x))`

Any feedback is much appreciated!

**Comment by brookeabrown** [comment]
Hello,
I am have the same question as the original post. Did you ever find a way to display the actual p value, instead of p < 2.2e-16?

Thank you!

**Comment by kassambara** [comment]
Yes — the `p < 2.2e-16` string is a hardcoded display floor. Two ways to show the real value:

**Option 1 — `p.accuracy`** removes the floor directly:

``` r
suppressWarnings(suppressPackageStartupMessages(library(ggpubr)))
set.seed(1); n <- 300
d <- data.frame(x = 1:n); d$y <- d$x + rnorm(n, sd = 60)
# Option 1: p.accuracy removes the hardcoded "p < 2.2e-16" string
ggscatter(d, x = "x", y = "y", add = "reg.line") + stat_cor(p.accuracy = 1e-120)
```

![](https://i.imgur.com/1iS5Rx6.png)

**Option 2 — compute + annotate** (best for a clean scientific label): read the exact p from `cor.test()` and place it yourself.

``` r
suppressWarnings(suppressPackageStartupMessages(library(ggpubr)))
set.seed(1); n <- 300
d <- data.frame(x = 1:n); d$y <- d$x + rnorm(n, sd = 60)
# Option 2 (clean scientific label for very small p): compute + annotate
pv  <- cor.test(d$x, d$y)$p.value                     # real p, e.g. 1.3e-76
lab <- sprintf("R = %.2f, p = %s", cor(d$x, d$y), formatC(pv, format = "e", digits = 1))
ggscatter(d, x = "x", y = "y", add = "reg.line") +
  annotate("text", x = -Inf, y = Inf, hjust = -0.1, vjust = 1.5, label = lab)
```

![](https://i.imgur.com/DS0vDvm.png)

Honest caveat: for p-values below machine epsilon (~2.2e-16), `scales::pvalue`/`format.pval` can produce an unwieldy long decimal, so for very small p the `annotate()` + `formatC(p, format="e")` route (Option 2) gives the tidiest label. Closing as resolved. 🙌
